### PR TITLE
Optimize dashboard query for loggedin user count for AB#16380

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#16380](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16380)

## Description
Reworked Blazor Admin to use DateOnly for dashboard query loggedin user counts
Updated Controller/Service to use DateOnly
Updated Delegate to use DateTimeOffset for query
Optimize GetLoggedInUsersCount (formerly GetDailyLoggedInUsersCount)
Added Indexes to UserProfile and UserProfileHistory for LastLoginDateTime
Cleared outstanding migration on seed data for ApplicationSettings
Add telemetry into Admin to aid in dev/test debugging

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## UI Changes
Faster

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Database/src/Delegates/DbProfileDelegate.cs [hunk 1]
```diff
@@ -251,20 +251,42 @@ public DbResult<List<UserProfile>> GetClosedProfiles(DateTime filterDateTime, in
...
+                });
 
-            return dateCount;
+            var userProfileHistoryQuery = this.dbContext.UserProfileHistory
+                .Where(u0 => u0.LastLoginDateTime >= startDateTimeOffset.UtcDateTime && u0.LastLoginDateTime <= endDateTimeOffset.UtcDateTime)
+                .Select(u0 => new
+                {
+                    u0.HdId,
+                    u0.LastLoginDateTime,
+                });
```
Comment (MikeLyttle) [inline Apps/Database/src/Delegates/DbProfileDelegate.cs:271]: Nit: the `u0` parameters can be swapped to `u` without conflicting with the parameters in the other query.

File: Apps/Admin/Server/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -65,7 +66,9 @@ public IActionResult GetRegisteredUserCount(int timeOffset)
         /// <summary>
         /// Retrieves the count of logged in user in the last day.
         /// </summary>
-        /// <param name="timeOffset">The current timezone offset from the client browser to UTC.</param>
+        /// <param name="startDateLocal">The starting date to get the user counts in the clients Local time.</param>
+        /// <param name="endDateLocal">The ending date for the query in the clients local time.</param>
```
Comment (MikeLyttle) [inline Apps/Admin/Server/Controllers/DashboardController.cs:70]:
```suggestion
        /// <param name="startDateLocal">The starting date to get the user counts in the client's local time.</param>
        /// <param name="endDateLocal">The ending date for the query in the client's local time.</param>
```

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Database/src/Delegates/DbProfileDelegate.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Database/src/Delegates/DbProfileDelegate.cs:271]: Yeah, it was an evolution but I've updated.

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]
I see what you mean by it looking like you're making multiple queries.

**Pull request merged**